<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Jarrod Daniels" />

<meta name="date" content="2020-11-16" />

<title>Lab 13: Network Analysis using Cytoscape and RCy3 (With PrettyDocs Theme)</title>

<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

/* A workaround for https://github.com/jgm/pandoc/issues/4278 */
a.sourceLine {
  pointer-events: auto;
}

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link rel="stylesheet" href="Lab13_files/style.css" type="text/css" />





</head>

<body>




<section class="page-header">
<h1 class="title toc-ignore project-name">Lab 13: Network Analysis using Cytoscape and RCy3 (With PrettyDocs Theme)</h1>
<h4 class="author project-author">Jarrod Daniels</h4>
<h4 class="date project-date">11/16/2020</h4>
</section>



<section class="main-content">
<div id="cytossape-tutorial-images" class="section level2">
<h2>Cytossape Tutorial Images</h2>
<div id="basic-data-visualization-tutorial" class="section level3">
<h3>Basic Data Visualization Tutorial</h3>
<p><img src="genomics-course/DataVis.png" /></p>
</div>
<div id="differentially-expressed-genes-tutorial" class="section level3">
<h3>Differentially Expressed Genes Tutorial</h3>
<p><img src="genomics-course/down.png" /></p>
<p><img src="genomics-course/data/ovariancancer.png" /></p>
</div>
</div>
<div id="overview-of-rcy3" class="section level2">
<h2>Overview of RCy3</h2>
<div id="installation-library-loading" class="section level3">
<h3>1 Installation &amp; Library Loading</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="cf">if</span>(<span class="op">!</span><span class="st">&quot;RCy3&quot;</span> <span class="op">%in%</span><span class="st"> </span><span class="kw">installed.packages</span>()){</a>
<a class="sourceLine" id="cb1-2" title="2">    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" title="3">    BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;RCy3&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" title="4">}</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(RCy3)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw">library</span>(prettydoc)</a></code></pre></div>
</div>
<div id="prerequisites" class="section level3">
<h3>2 Prerequisites</h3>
<p>In addition to this package (RCy3), you will need:</p>
<ul>
<li><strong>Cytoscape 3.6.1</strong> or greater, which can be downloaded from <a href="http://www.cytoscape.org/download.php" class="uri">http://www.cytoscape.org/download.php</a>. Simply follow the installation instructions on screen.</li>
</ul>
</div>
<div id="getting-started" class="section level3">
<h3>3 Getting started</h3>
<p>First, launch Cytoscape and keep it running whenever using RCy3. Confirm that you have everything installed and running:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw">cytoscapePing</span> ()</a></code></pre></div>
<pre><code>## [1] &quot;You are connected to Cytoscape!&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw">cytoscapeVersionInfo</span> ()</a></code></pre></div>
<pre><code>##       apiVersion cytoscapeVersion 
##             &quot;v1&quot;          &quot;3.8.2&quot;</code></pre>
</div>
<div id="my-first-network" class="section level3">
<h3>4 My first network</h3>
<p>Let’s create a Cytoscape network from some basic R objects</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">nodes &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">id=</span><span class="kw">c</span>(<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 1&quot;</span>,<span class="st">&quot;node 2&quot;</span>,<span class="st">&quot;node 3&quot;</span>),</a>
<a class="sourceLine" id="cb7-2" title="2">           <span class="dt">group=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;B&quot;</span>), <span class="co"># categorical strings</span></a>
<a class="sourceLine" id="cb7-3" title="3">           <span class="dt">score=</span><span class="kw">as.integer</span>(<span class="kw">c</span>(<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">15</span>,<span class="dv">5</span>)), <span class="co"># integers</span></a>
<a class="sourceLine" id="cb7-4" title="4">           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb7-5" title="5">edges &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">source=</span><span class="kw">c</span>(<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 2&quot;</span>),</a>
<a class="sourceLine" id="cb7-6" title="6">           <span class="dt">target=</span><span class="kw">c</span>(<span class="st">&quot;node 1&quot;</span>,<span class="st">&quot;node 2&quot;</span>,<span class="st">&quot;node 3&quot;</span>,<span class="st">&quot;node 3&quot;</span>),</a>
<a class="sourceLine" id="cb7-7" title="7">           <span class="dt">interaction=</span><span class="kw">c</span>(<span class="st">&quot;inhibits&quot;</span>,<span class="st">&quot;interacts&quot;</span>,<span class="st">&quot;activates&quot;</span>,<span class="st">&quot;interacts&quot;</span>),  <span class="co"># optional</span></a>
<a class="sourceLine" id="cb7-8" title="8">           <span class="dt">weight=</span><span class="kw">c</span>(<span class="fl">5.1</span>,<span class="fl">3.0</span>,<span class="fl">5.2</span>,<span class="fl">9.9</span>), <span class="co"># numeric</span></a>
<a class="sourceLine" id="cb7-9" title="9">           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb7-10" title="10"></a>
<a class="sourceLine" id="cb7-11" title="11"><span class="kw">createNetworkFromDataFrames</span>(nodes,edges, <span class="dt">title=</span><span class="st">&quot;my first network&quot;</span>, <span class="dt">collection=</span><span class="st">&quot;DataFrame Example&quot;</span>)</a></code></pre></div>
<p>Output in Cytoscape:</p>
<p><img src="genomics-course/images/exampleNetwork.JPG" /></p>
<div id="switch-styles" class="section level4">
<h4>4.1 Switch styles</h4>
<p>Check out the marquee style!</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"><span class="kw">setVisualStyle</span>(<span class="st">&#39;Marquee&#39;</span>)</a></code></pre></div>
<pre><code>##                 message 
## &quot;Visual Style applied.&quot;</code></pre>
<p>Output in Cytoscape:</p>
<p><img src="genomics-course/images/Example2.JPG" /></p>
</div>
<div id="my-own-style" class="section level4">
<h4>4.2 My own style</h4>
<p>Create your own style with node attribute fill mappings and some defaults</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">style.name =<span class="st"> &quot;myStyle&quot;</span></a>
<a class="sourceLine" id="cb10-2" title="2">defaults &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">NODE_SHAPE=</span><span class="st">&quot;diamond&quot;</span>,</a>
<a class="sourceLine" id="cb10-3" title="3">                 <span class="dt">NODE_SIZE=</span><span class="dv">30</span>,</a>
<a class="sourceLine" id="cb10-4" title="4">                 <span class="dt">EDGE_TRANSPARENCY=</span><span class="dv">120</span>,</a>
<a class="sourceLine" id="cb10-5" title="5">                 <span class="dt">NODE_LABEL_POSITION=</span><span class="st">&quot;W,E,c,0.00,0.00&quot;</span>)</a>
<a class="sourceLine" id="cb10-6" title="6">nodeLabels &lt;-<span class="st"> </span><span class="kw">mapVisualProperty</span>(<span class="st">&#39;node label&#39;</span>,<span class="st">&#39;id&#39;</span>,<span class="st">&#39;p&#39;</span>)</a>
<a class="sourceLine" id="cb10-7" title="7">nodeFills &lt;-<span class="st"> </span><span class="kw">mapVisualProperty</span>(<span class="st">&#39;node fill color&#39;</span>,<span class="st">&#39;group&#39;</span>,<span class="st">&#39;d&#39;</span>,<span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;#FF9900&quot;</span>,<span class="st">&quot;#66AAAA&quot;</span>))</a>
<a class="sourceLine" id="cb10-8" title="8">arrowShapes &lt;-<span class="st"> </span><span class="kw">mapVisualProperty</span>(<span class="st">&#39;Edge Target Arrow Shape&#39;</span>,<span class="st">&#39;interaction&#39;</span>,<span class="st">&#39;d&#39;</span>,<span class="kw">c</span>(<span class="st">&quot;activates&quot;</span>,<span class="st">&quot;inhibits&quot;</span>,<span class="st">&quot;interacts&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;Arrow&quot;</span>,<span class="st">&quot;T&quot;</span>,<span class="st">&quot;None&quot;</span>))</a>
<a class="sourceLine" id="cb10-9" title="9">edgeWidth &lt;-<span class="st"> </span><span class="kw">mapVisualProperty</span>(<span class="st">&#39;edge width&#39;</span>,<span class="st">&#39;weight&#39;</span>,<span class="st">&#39;p&#39;</span>)</a>
<a class="sourceLine" id="cb10-10" title="10"></a>
<a class="sourceLine" id="cb10-11" title="11"><span class="kw">createVisualStyle</span>(style.name, defaults, <span class="kw">list</span>(nodeLabels,nodeFills,arrowShapes,edgeWidth))</a>
<a class="sourceLine" id="cb10-12" title="12"><span class="kw">setVisualStyle</span>(style.name)</a></code></pre></div>
<p>Output in Cytoscape:</p>
<p><img src="genomics-course/images/Example3.JPG" /></p>
<p>Pro-tip: if you want to set NODE_WIDTH and NODE_HEIGHT independently, you also need to unlock the node dimensions with…</p>
</div>
</div>
<div id="bioconductor-graph-example" class="section level3">
<h3>5 Bioconductor graph example</h3>
<p>Alternatively, you might want to start from a Bioconductor graphNEL object. Here we create a 4-node graph in R, send it to Cytoscape for display and layout. For the sake of simplicity, no node attributes and no visual styles are included; those topics are covered in subsequent steps.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">    g =<span class="st"> </span><span class="kw">new</span> (<span class="st">&#39;graphNEL&#39;</span>, <span class="dt">edgemode=</span><span class="st">&#39;directed&#39;</span>)</a>
<a class="sourceLine" id="cb11-2" title="2">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw">addNode</span> (<span class="st">&#39;A&#39;</span>, g)</a>
<a class="sourceLine" id="cb11-3" title="3">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw">addNode</span> (<span class="st">&#39;D&#39;</span>, g)</a>
<a class="sourceLine" id="cb11-4" title="4">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw">addNode</span> (<span class="st">&#39;C&#39;</span>, g, <span class="dt">edges =</span> <span class="kw">list</span>(<span class="st">&#39;D&#39;</span>))</a>
<a class="sourceLine" id="cb11-5" title="5">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw">addNode</span> (<span class="st">&#39;B&#39;</span>, g, <span class="dt">edges =</span> <span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&#39;A&#39;</span>,<span class="st">&#39;D&#39;</span>,<span class="st">&#39;C&#39;</span>)))</a>
<a class="sourceLine" id="cb11-6" title="6">    <span class="kw">createNetworkFromGraph</span> (g, <span class="dt">title=</span><span class="st">&#39;simple network&#39;</span>, <span class="dt">collection=</span><span class="st">&#39;GraphNEL Example&#39;</span>)</a></code></pre></div>
<p>This output is the same as our graph from part 1</p>
<p><img src="genomics-course/images/exampleNetwork.JPG" /></p>
<div id="add-node-attributes" class="section level4">
<h4>5.1 Add node attributes</h4>
<p>We often know quite a lot about the nodes and edges in our graphs. By conveying this information visually, the graph will be easier to explore. For instance, we may know that protein A phosphorylates protein B, that A is a kinase and B a transcription factor, and that their mRNA expression (compared to a control) is a log2 fold change of 1.8 and 3.2 respectively. One of the core features of Cytoscape is visual styles, which allow you to specify how data values (e.g., kinase’,transcription factor’; expression ratios) should be conveyed in the visual properties of the graph (e.g., node shape, node color or size).</p>
<p>We continue with the simple 4-node graph, adding two kinds data values (moleculeType’ andlog2fc’). The easiest way to do this is via data.frames. However, you can also include attributes together with the original graph models as Bioconductor graphs, igraphs or data.frames and then use the provided create functions to create and load in a single step (see createNetworkFromGraph, createNetworkFromIgraph and createNetworkFromDataFrames functions). Check out the other vignettes for more exampls.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">    df &lt;-<span class="st"> </span><span class="kw">data.frame</span> (<span class="dt">moleculeType=</span><span class="kw">c</span>(<span class="st">&#39;kinase&#39;</span>,<span class="st">&#39;TF&#39;</span>,<span class="st">&#39;cytokine&#39;</span>,<span class="st">&#39;cytokine&#39;</span>),</a>
<a class="sourceLine" id="cb12-2" title="2">                     <span class="dt">log2fc=</span><span class="kw">c</span>(<span class="fl">1.8</span>,<span class="fl">3.0</span>,<span class="op">-</span><span class="fl">1.2</span>,<span class="op">-</span><span class="fl">2.5</span>),</a>
<a class="sourceLine" id="cb12-3" title="3">                     <span class="dt">row.names =</span> <span class="kw">c</span>(<span class="st">&#39;A&#39;</span>,<span class="st">&#39;B&#39;</span>,<span class="st">&#39;C&#39;</span>,<span class="st">&#39;D&#39;</span>), <span class="co"># row.names = node names</span></a>
<a class="sourceLine" id="cb12-4" title="4">                     <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)       <span class="co"># important when loading strings!</span></a>
<a class="sourceLine" id="cb12-5" title="5">    <span class="kw">loadTableData</span> (df)</a></code></pre></div>
<pre><code>## [1] &quot;Success: Data loaded in defaultnode table&quot;</code></pre>
<p>Note that adding the attributes does not in itself cause the appearance of the graph to change. Such a change requires that you specify and apply visual style mappings, which will be explained in the next section. You can, however, examine these attributes in Cytoscape, using Cytoscape’s the Data Panel to display data values associated with selected nodes immediately below the Cytoscape window.</p>
</div>
<div id="modifying-the-display-defualts-and-mappings" class="section level4">
<h4>5.2 Modifying the display: defualts and mappings</h4>
<p>RCy3 provides an easy way to not only change the default styles, but more interestingly, RCy3 also provides easy access to mapping your data to visual styles, e.g., allowing the size, shape and color of nodes and edges to be determined by the data you have associated with those nodes and edges.</p>
<p>First, let’s change the the defaults.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1">    <span class="kw">setNodeShapeDefault</span> (<span class="st">&#39;OCTAGON&#39;</span>)</a>
<a class="sourceLine" id="cb14-2" title="2">    <span class="kw">setNodeColorDefault</span> (<span class="st">&#39;#AAFF88&#39;</span>)</a>
<a class="sourceLine" id="cb14-3" title="3">    <span class="kw">setNodeSizeDefault</span>  (<span class="dv">60</span>)</a>
<a class="sourceLine" id="cb14-4" title="4">    <span class="kw">setNodeFontSizeDefault</span> (<span class="dv">30</span>)</a></code></pre></div>
<p>Output in Cytoscape:</p>
<p><img src="genomics-course/images/Example4.JPG" /></p>
<p>Now we will add some visual mappings. Let’s map `moleculeType’ to node shapes. First, we can see which shapes are available in Cytoscape, then we can define the mapping with paired lists.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1">    <span class="kw">getNodeShapes</span> ()   <span class="co"># diamond, ellipse, trapezoid, triangle, etc.</span></a></code></pre></div>
<pre><code>## [1] &quot;TRIANGLE&quot;        &quot;OCTAGON&quot;         &quot;RECTANGLE&quot;       &quot;ROUND_RECTANGLE&quot;
## [5] &quot;PARALLELOGRAM&quot;   &quot;ELLIPSE&quot;         &quot;DIAMOND&quot;         &quot;VEE&quot;            
## [9] &quot;HEXAGON&quot;</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">    column &lt;-<span class="st"> &#39;moleculeType&#39;</span></a>
<a class="sourceLine" id="cb17-2" title="2">    values &lt;-<span class="st"> </span><span class="kw">c</span> (<span class="st">&#39;kinase&#39;</span>,  <span class="st">&#39;TF&#39;</span>,<span class="st">&#39;cytokine&#39;</span>)</a>
<a class="sourceLine" id="cb17-3" title="3">    shapes &lt;-<span class="st"> </span><span class="kw">c</span> (<span class="st">&#39;DIAMOND&#39;</span>, <span class="st">&#39;TRIANGLE&#39;</span>, <span class="st">&#39;RECTANGLE&#39;</span>)</a>
<a class="sourceLine" id="cb17-4" title="4">    <span class="kw">setNodeShapeMapping</span> (column, values, shapes)</a></code></pre></div>
<pre><code>## style.name not specified; updating &quot;default&quot; style.</code></pre>
<p><img src="genomics-course/images/Example5.JPG" /></p>
<p>The node shape mapping is an example of a discrete mapping, where a style is defined for each, discrete value. This is useful for categorical data (like type) where there is only a limited set of possible values. This is in contast to the other two other types of mappings: continuous and passthrough. In the case of expression values, for example, we will want to use continuous mapping (e.g., to node color), defining a small set of control points, rather than an explicit color for each possible data value. Cytoscape will simply interpolate between the control points to provide a gradient of colors. Let’s try that one now</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1">    column &lt;-<span class="st"> &#39;log2fc&#39;</span></a>
<a class="sourceLine" id="cb19-2" title="2">    control.points &lt;-<span class="st"> </span><span class="kw">c</span> (<span class="op">-</span><span class="fl">3.0</span>, <span class="fl">0.0</span>, <span class="fl">3.0</span>)</a>
<a class="sourceLine" id="cb19-3" title="3">    colors &lt;-<span class="st">  </span><span class="kw">c</span> (<span class="st">&#39;#5588DD&#39;</span>, <span class="st">&#39;#FFFFFF&#39;</span>, <span class="st">&#39;#DD8855&#39;</span>)</a>
<a class="sourceLine" id="cb19-4" title="4">    <span class="kw">setNodeColorMapping</span> (column, control.points, colors)</a></code></pre></div>
<pre><code>## style.name not specified; updating &quot;default&quot; style.</code></pre>
<p>Output in Cytoscape:</p>
<p><img src="genomics-course/images/Example6.JPG" /></p>
<p>Note that there are three colors and three control points. However, you can also specify two additional colors beyond the number of control points if you want to set extreme (or out-of-bounds) colors for values less than or greater than your control points.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1">    control.points &lt;-<span class="st"> </span><span class="kw">c</span> (<span class="op">-</span><span class="fl">2.0</span>, <span class="fl">0.0</span>, <span class="fl">2.0</span>)</a>
<a class="sourceLine" id="cb21-2" title="2">    colors &lt;-<span class="st">  </span><span class="kw">c</span> (<span class="st">&#39;#2255CC&#39;</span>, <span class="st">&#39;#5588DD&#39;</span>, <span class="st">&#39;#FFFFFF&#39;</span>, <span class="st">&#39;#DD8855&#39;</span>,<span class="st">&#39;#CC5522&#39;</span>)</a>
<a class="sourceLine" id="cb21-3" title="3">    <span class="kw">setNodeColorMapping</span> (column, control.points, colors)</a></code></pre></div>
<pre><code>## style.name not specified; updating &quot;default&quot; style.</code></pre>
<p><img src="genomics-course/images/Example7.JPG" /></p>
<p>Now, add a node size rule, using log2fc again as controlling node values.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1">control.points =<span class="st"> </span><span class="kw">c</span> (<span class="op">-</span><span class="fl">3.0</span>, <span class="fl">2.0</span>, <span class="fl">3.0</span>)</a>
<a class="sourceLine" id="cb23-2" title="2">    sizes     =<span class="st"> </span><span class="kw">c</span> (<span class="dv">20</span>, <span class="dv">80</span>, <span class="dv">90</span>)</a>
<a class="sourceLine" id="cb23-3" title="3">    <span class="kw">setNodeSizeMapping</span> (column, control.points, sizes)</a></code></pre></div>
<pre><code>## style.name not specified; updating &quot;default&quot; style.</code></pre>
<p><img src="genomics-course/images/Example8.JPG" /></p>
<p>If you recall the third type of mapping, passthrough, we can see it already working in our current network example. The node labels! By default, the name column is mapped to the node label property using passthrough logic: the value is passed directly to the style property.</p>
</div>
<div id="section" class="section level4">
<h4>5.3</h4>
<p>Let us now try selecting nodes in Cytoscape from R. Select the C node by name:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1"><span class="kw">selectNodes</span> (<span class="st">&#39;C&#39;</span>,<span class="st">&#39;name&#39;</span>)</a></code></pre></div>
<pre><code>## $nodes
## [1] 160
## 
## $edges
## list()</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1"><span class="kw">getSelectedNodes</span> ()</a></code></pre></div>
<pre><code>## [1] &quot;C&quot;</code></pre>
<p><img src="genomics-course/images/Example9.JPG" /></p>
<p>Now we wish to extend the selected nodes to include the first neighbors of the already-selected node B. This is a common operation: for instance, after selecting one or more nodes based on experimental data or annotation, you may want to explore these in the context of interaction partners (in a protein-protein network) or in relation to upstream and downstream partners in a signaling or metabolic network. Type:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" title="1"><span class="kw">selectFirstNeighbors</span> ()</a></code></pre></div>
<pre><code>## $nodes
## [1] 159 160 161
## 
## $edges
## list()</code></pre>
<p><img src="genomics-course/images/Example10.JPG" /></p>
<p>You will see that three nodes are now selected. Get their names back to R as a list:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" title="1">node.names &lt;-<span class="st"> </span><span class="kw">getSelectedNodes</span> ()</a></code></pre></div>
<p>And, finally, deselection works as you’d expect by means of a general clearSelection function:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1"><span class="kw">clearSelection</span>()</a>
<a class="sourceLine" id="cb32-2" title="2">?clearSelection</a></code></pre></div>
</div>
<div id="saving-and-export" class="section level4">
<h4>5.4 Saving and export</h4>
<p>Session files save everything. As with most project software, we recommend saving often!</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" title="1"><span class="kw">saveSession</span>(<span class="st">&#39;vignette_session&#39;</span>) <span class="co">#.cys</span></a></code></pre></div>
<pre><code>## Warning: This file has been overwritten.</code></pre>
<p>Note: If you don’t specify a complete path, the files will be saved relative to your current working directory in R.</p>
<div id="saving-high-resolution-image-files" class="section level5">
<h5>5.4.1 Saving high resolution image files</h5>
<p>You can export extremely high resolution images, including vector graphic formats.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" title="1">    full.path=<span class="kw">paste</span>(<span class="kw">getwd</span>(),<span class="st">&#39;vignette_image&#39;</span>,<span class="dt">sep=</span><span class="st">&#39;/&#39;</span>)</a>
<a class="sourceLine" id="cb35-2" title="2">    <span class="kw">exportImage</span>(full.path, <span class="st">&#39;PNG&#39;</span>, <span class="dt">zoom=</span><span class="dv">200</span>) <span class="co">#.png scaled by 200%</span></a>
<a class="sourceLine" id="cb35-3" title="3">    <span class="kw">exportImage</span>(full.path, <span class="st">&#39;PDF&#39;</span>) <span class="co">#.pdf</span></a>
<a class="sourceLine" id="cb35-4" title="4">    ?exportImage</a></code></pre></div>
<p>This marks the end of the graphical introduction portion of the vignette as well as the end of this lab. Thank you for this semester! I understand it has been a really difficult past few months for all of us and I wanted to personally thank you for being so accomodating with the class.</p>
</div>
</div>
</div>
</div>
</section>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
